roles/openshift_clock_facts/tasks/main.yaml

```diff
@@ -0,0 +1,40 @@
+---
+- fail:
+    msg: "openshift_clock_method must be set to either 'none', 'auto', 'ntpd', or 'chrony'"
+  when:  openshift_clock_method is defined and
+    openshift_clock_method not in ["none", "auto", "ntpd", "chrony"]
+
+- set_fact:
+    openshift_clock_method: chrony
+  when: openshift_clock_method is defined and openshift_clock_method == "auto" and
+    openshift.common.is_containerized and openshift.common.is_atomic
+
+- set_fact:
+    openshift_clock_method: ntpd
+  when: openshift_clock_method is defined and openshift_clock_method == "auto"
```

Comment:
This will probably not work as intended. Ansible doesn't allow for re-assigning variables that have already been set. You can only reliably do this if testing that the variable you are setting with set_fact `is not defined`.

Instead, you will probable want to just persist 'auto' to openshift_facts, test that for the invalid conditions, and then set the dependency role variables based on something like this:
```
when: openshift.clock.method is defined and (openshift.clock.method == 'ntpd' or (openshift.clock.method == 'auto'  and default_clock_method == 'ntpd'))
```